rfcs/text/0000-roadmaps+planning.md

```diff
@@ -0,0 +1,421 @@
+
+It is strongly RECOMMENDED that engineering teams buffer a **minimum of 15%** capacity for responding to emerging issues during a Release, supporting DCP community reviews for RFC(s), time-off, regular DCP meetings, transitioning developers on or off a team, etc.
+
+#### Modeling Objectives Requiring Further Research
+
+If an Objective requires further exploration or design before it can be estimated and implemented, then its Product Owner MUST:
+
+1. Create a new ZenHub issue and label it as a Spike.
+1. Assign one (and only one) Owner responsible for completing the Spike.
+1. Document and monitor the *work-in-progress*. If an RFC is required, a reference to the draft RFC in the *dcp-community* repo is required; otherwise a reference to other documentation is appropriate.
+(For designs, it is RECOMMENDED that a temporary working group channel be created in *HumanCellAtlas* Slack and documented in the Spike to ensure that the *state of play* is visible to the DCP community.)
+1. Document its Acceptance Criteria such as an approved RFC. 
+1. Add the Spike to its related Epic.
+1. Add the Spike as a blocking dependency in its related Epic.
+1. Assign the Spike to a specific Milestone in the Release. Spikes have a finite duration.
+
+The related Epic cannot be estimated and refined until the completion of its blocking Spike. Once the Spike is closed, then its Epic MUST be modeled as outlined in [Creating the Engineering Plan](#creating-the-engineering-plan).
+
+
+#### Modeling Objectives that Require Multiple Releases
+
+Objectives which span multiple Releases MUST be decomposed into one or more Epics per Release. The Epics MUST be assigned to a Release and linked as blocking dependencies. For example, *Epic 1 in Release 1 blocks Epic 2 in Release 2 blocks Epic 3 in Release 3* which clearly documents their dependency chain and schedule.
+
+An RFC is strongly RECOMMENDED for Objectives that require multiple releases to complete.
+
+#### Creating the Engineering Plan
+[Creating the Engineering Plan]:#creating-the-engineering-plan
+
+***Next Release Milestone 1 Week 2***. In consultation with their relevant Technical Leads and collaborators, the Product Owner MUST complete the refinement and modeling for *well-understood* Epics, including the addition of child issues for product readiness and the surfacing of dependencies between software components using ZenHub Dependencies.
+
+Milestones MUST be assigned to all issues assigned to the Release not blocked by a Spike.
+
+It is RECOMMENDED that caution be exercised when scheduling issues in a dependency chain in the same Milestone. Historically, this has been the source of slips in DCP schedules.
+
+The Release Shepherd MUST validate accurate modeling of Objectives for the Release in ZenHub and create a Release epic named “Q#YYYY” such as “Q42019” which contains all the related *dcp* Epics and issues for the Release. This will automatically include all child issues in other component repo(s).
```

Comment:
Just a small question - if the releases are named Q#YYYY, two years from now will they be sorted like this in the list of releases:

Q12019
Q12020
Q12021
Q22019
Q22020
Q32019
Q32020

etc? Or maybe releases are removed from the list after they have passed? If not, I imagine it'd be preferable to have them sorted by year, so YYYYQ#.

Reply:
RE _hope that themes are pretty much the only DCP level epics_, I expect this to be mostly true when there are **annual** roadmaps in 2020. The exceptions would be quarterly technical objectives and emerging high-priority DevSecOps issues. For 2019, we still would want to include high visibility (but non-roadmap) epics like your pipelines.

RE releases -I have not reviewed the documentation, but it appears that the Releases and Milestones are ordered by due date in the drop down. So expired items appear first. Nonetheless, I'd be happy to adopt `YYYYQ#`. I suspect that we would want to delete Releases to avoid unwieldy lists.

Reply:
If we delete releases do we lose any of the reporting information that Zenhub has?

Reply:
BTW - the naming above is completely unrelated to Release names. It's the name of an Epic but it does suggest that we should have templates for both Releases and Milestones in ZenHub so thanks for the suggestion. 

RE _delete releases_ - I would expect the loss of reporting information so we would want to ensure that the releases are done and any information has been summarized in a Retrospective.

Reply:
Oh got it. I missed that detail - if it's the name of the epic I am not as phased on the name template. And if we plan to delete releases after the fact, I'm not worried about having a name template there either.

Reply:
We can either _Close_ or _Delete_ both Releases and Milestones. The dropdown for selecting Releases and Milestones has tabs for _Open_ and _Closed_.